src/harness/request_handler.py

```diff
@@ -127,12 +131,23 @@ def _Request(url, request, config, is_post_method):
     logging.info('POST Request to URL %s :\n%s', url, request)
   else:
     logging.info('GET Request to URL %s', url)
-  try:
-    conn.perform()
-  except pycurl.error as e:
-    # e contains a tuple (libcurl_error_code, string_description).
-    # See https://curl.haxx.se/libcurl/c/libcurl-errors.html
-    raise CurlError(e.args[1], e.args[0])
+
+  for attempt_count in range(MAX_REQUEST_ATTEMPT_COUNT):
+    try:
+      conn.perform()
+      error_message = ''
+      break
+    except pycurl.error as e:
+      # e contains a tuple (libcurl_error_code, string_description).
+      # See https://curl.haxx.se/libcurl/c/libcurl-errors.html
+      error_message += ' %s\n' % str(CurlError(e.args[1], e.args[0]))
+      time.sleep(REQUEST_ATTEMPT_DELAY_SECOND)
+    except Exception as e:
+      error_message += ' %s\n' % str(e)
+      time.sleep(REQUEST_ATTEMPT_DELAY_SECOND)
+  if error_message:
+    logging.error(error_message.strip())
+    raise
```

Comment:
Did you run this? I think you need to call `raise` from within the `except`. Here's an experiment I did:

```
In [5]: try:
   ...:     raise ImportError("test")
   ...: except Exception as e:
   ...:     print "Inside the except: ", e
   ...: raise
   ...: 
   ...: 
Inside the except:  test
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-7fa1da41e333> in <module>()
      3 except Exception as e:
      4     print "Inside the except: ", e
----> 5 raise
      6 

TypeError: exceptions must be old-style classes or derived from BaseException, not NoneType
```

Reply:
Strange -- I'm using 2.7.15 (from iPython) and I wouldn't have expected such a large difference between minor versions.

Depending on where you look, our repo says we use 2.7 [here](https://github.com/Wireless-Innovation-Forum/Spectrum-Access-System) or 2.7.12 [here](https://github.com/Wireless-Innovation-Forum/Spectrum-Access-System/tree/master/src/harness/). I'm not sure what flavor ITS is using.

Reply:
Tried on Python 2.7.15
`Python 2.7.15 (default, Aug 17 2018, 22:39:05)
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] on darwin`

I am getting the same output 😬

Reply:
No, it is not, maybe try creating a py file? If it is still not working, we may have to think about another logic to implement it.

Reply:
Hmm, very strange! That worked, and I ran an example test case (REG_6) to double-check. Thanks for the tip and for the PR itself!